math/mode.go

```diff
@@ -0,0 +1,61 @@
+// mode.go
+// author(s): [CalvinNJK] (https://github.com/CalvinNJK)
+// description: Finding Mode Value In an Array
+// see mode.go
+
+package math
+
+//package constraints
+
+import (
+	"errors"
+
+	"github.com/TheAlgorithms/Go/constraints"
+)
+
+type Number interface {
+	constraints.Float | constraints.Integer
+}
+
+// Error when array is empty
+var emptyModeArr = errors.New("Array is empty")
```

Comment:
This error should be constant.

```suggestion
var ErrEmptySlice = errors.New("empty slice provided")
```

PS: In Go, errors string don't start with a capital letter